services/grpc/internal/transformations/graph_objects.go

```diff
@@ -0,0 +1,36 @@
+package transformations
+
+type Node struct {
+	NodeId string `db:"nodeId"`
+	Doc string `db:"doc"`
+
+	EdgesOut string[] `db:"edgesOut"` // IDs of edges whose destination is this node
+	EdgesIn string[] `db:"edgesIn"`   // ID of edges whose source is this node
+
+	// Metadata
+	Date string `db:"date"`
+	Author string `db:"author"`
+}
+
+type Edge struct {
+	EdgeId string `db:"edgeId"`
+	SourceDoc string `db:"sourceDoc"`
+	TargetDoc string `db:"targetDoc`
+	Args FxnArgs `db:"args"` 				// f(SourceDoc, Args) -> TargetDoc
+	InverseArgs FxnArgs `db:"inverseArgs"`  // f(TargetDoc, InverseArgs) -> SourceDoc
+
+	SourceNodeId string `db:"sourceNodeId"`
+	TargetNodeId string `db:"targetNodeId"`
+
+	// Metadata
+	Date string `db:"date"`
+	Author string `db:"author"`
+}
+
+type Scope struct {
```

Comment:
A Scope should probably have some kind of metadata object (i.e. title, description, etc).

Reply:
I do like the idea of somehow defining a Scope _in terms_ of Nodes and Edges. In the simplest case, over time a Scope metadata could be modified and it could be nice to preserve that history. In a more complex / aspirational case perhaps we could apply set operations / transformations to the _contents_ of Scopes and want to preserve _that_ history.

Reply:
TODO: Add a comment/TODO to think about how to represent a Scope in terms of Nodes/Edges.